src/client/containers/chart-detail-page/chart-detail-useeffect.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import ChartDetailSmartPageHumidity from './chart-detailsmartpage-humidity';
+import { getTokenWithHeaders } from '../../firebase/getTokenWithHeaders';
+import { ChartDataContext } from './chart-detail-context';
+
+const ChartDetailsWithCropData = () => {
+  const [boundary, setBoundary] = useState({});
+  const [materialName, setMaterialName] = useState('');
+  const [materialId, setMaterialId] = useState('');
+  const [sensorData, setSensorData] = useState([]);
+  const [active, setActive] = useState(false);
+  const [startDate, setStartDate] = useState('');
+  const [currentDate, setCurrentDate] = useState('');
+  const [stages, setStages] = useState([]);
+  const [units, setUnits] = useState('');
+
+  const buttonClick = (e) => {
+    e.preventDefault();
+    setActive(!active);
+  };
+  const handleClick = (e) => {
+    setMaterialId(e.target.id);
+    setMaterialName(e.target.innerText);
+  };
+
+  // useEffect for optimalvalues
+  useEffect(() => {
+    async function fetchBatchStagesDefaultData() {
+      try {
+        const header = await getTokenWithHeaders();
+        const getCropStageValue = await fetch(
+          'api/batch-default-values/1?stage=current',
```

Comment:
`fk_crop_id` has changed in ` crop_stages ` table but  we are using endpoint may be you need update your database and check the endpoint so that it will work with `api/batch-default-values/1?stage=current`  bcz we need to show the current stage optimal values

Reply:
Got it, I'll try after I update it 👍